cmd/ci-op-configs-mirror/main.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/interrupts"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+)
+
+const (
+	prowgenConfigFile = ".config.prowgen"
+)
+
+type options struct {
+	dryRun         bool
+	configDir      string
+	destinationDir string
+	fromOrg        string
+	toOrg          string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configDir, "config-path", "", "Path to directory containing ci-operator configurations")
+	fs.StringVar(&o.destinationDir, "destination-path", "", "Path of the destination of the generated ci-operator configurations")
+
+	fs.StringVar(&o.fromOrg, "from-org", "", "Mirror the ci-operator configuration files from a single organization, if specified")
+	fs.StringVar(&o.toOrg, "to-org", "", "Name of the organization which the ci-operator configuration files will be mirrored")
+
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Whether to actually generate the new ci-operator configuration files")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func (o *options) validate() error {
+	if len(o.configDir) == 0 {
+		return errors.New("--config-path is not defined")
+	}
+	if len(o.destinationDir) == 0 && !o.dryRun {
+		return errors.New("--destination-path is not defined")
+	}
+	if len(o.toOrg) == 0 {
+		return errors.New("--to-org is not defined")
+	}
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid option")
+	}
+
+	go func() {
+		interrupts.WaitForGracefulShutdown()
+		os.Exit(1)
+	}()
+
+	prowgenConfig := config.Prowgen{Private: true}
+
+	dryBuff := &bytes.Buffer{}
+	callback := func(rbc *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
+		logger := logrus.WithFields(logrus.Fields{"org": repoInfo.Org, "repo": repoInfo.Repo})
+
+		if len(o.fromOrg) > 0 && repoInfo.Org != o.fromOrg {
```

Comment:
I don't think this is the correct selection logic. We don't want to mirror `openshift/ci-tools` for instance.

Reply:
In that case, I think it's better to add an `--ignore-repo` flag.

Reply:
I think we should make our lives simpler (by behaving consistently everywhere) here and just use the same logic like in config-brancher or private-org-sync and by default include all configs that promote official images.

Possibly only provide the options to limit that set to a single repo or an org, but I don't think that's necessary.